pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -339,33 +339,42 @@ void writeMetadata()
     ColumnIndexCreationInfo timeColumnIndexCreationInfo = indexCreationInfoMap.get(timeColumn);
     if (timeColumnIndexCreationInfo != null) {
+      long startTime;
+      long endTime;
+      TimeUnit timeUnit;
+
       // Use start/end time in config if defined
       if (config.getStartTime() != null) {
-        checkTime(config, config.getStartTime(), config.getEndTime(), segmentName);
-        properties.setProperty(SEGMENT_START_TIME, config.getStartTime());
-        properties.setProperty(SEGMENT_END_TIME, Preconditions.checkNotNull(config.getEndTime()));
-        properties.setProperty(TIME_UNIT, Preconditions.checkNotNull(config.getSegmentTimeUnit()));
+        startTime = Long.parseLong(config.getStartTime());
+        endTime = Long.parseLong(config.getEndTime());
+        timeUnit = Preconditions.checkNotNull(config.getSegmentTimeUnit());
       } else {
-        Object minTime = Preconditions.checkNotNull(timeColumnIndexCreationInfo.getMin());
-        Object maxTime = Preconditions.checkNotNull(timeColumnIndexCreationInfo.getMax());
+        String startTimeStr = timeColumnIndexCreationInfo.getMin().toString();
+        String endTimeStr = timeColumnIndexCreationInfo.getMax().toString();
 
         if (config.getTimeColumnType() == SegmentGeneratorConfig.TimeColumnType.SIMPLE_DATE) {
           // For TimeColumnType.SIMPLE_DATE_FORMAT, convert time value into millis since epoch
           DateTimeFormatter dateTimeFormatter = DateTimeFormat.forPattern(config.getSimpleDateFormat());
-          final long minTimeMillis = dateTimeFormatter.parseMillis(minTime.toString());
-          final long maxTimeMillis = dateTimeFormatter.parseMillis(maxTime.toString());
-          checkTime(config, minTimeMillis, maxTimeMillis, segmentName);
-          properties.setProperty(SEGMENT_START_TIME, minTimeMillis);
-          properties.setProperty(SEGMENT_END_TIME, maxTimeMillis);
-          properties.setProperty(TIME_UNIT, TimeUnit.MILLISECONDS);
+          startTime = dateTimeFormatter.parseMillis(startTimeStr);
+          endTime = dateTimeFormatter.parseMillis(endTimeStr);
+          timeUnit = TimeUnit.MILLISECONDS;
         } else {
           // by default, time column type is TimeColumnType.EPOCH
-          checkTime(config, minTime, maxTime, segmentName);
-          properties.setProperty(SEGMENT_START_TIME, minTime);
-          properties.setProperty(SEGMENT_END_TIME, maxTime);
-          properties.setProperty(TIME_UNIT, Preconditions.checkNotNull(config.getSegmentTimeUnit()));
+          startTime = Long.parseLong(startTimeStr);
+          endTime = Long.parseLong(endTimeStr);
+          timeUnit = Preconditions.checkNotNull(config.getSegmentTimeUnit());
         }
       }
+
+      if (!config.isSkipTimeValueCheck()) {
+        Interval timeInterval = new Interval(timeUnit.toMillis(startTime), timeUnit.toMillis(endTime));
+        Preconditions.checkState(TimeUtils.isValidTimeInterval(timeInterval),
```

Comment:
I am not sure how exactly Interval will be printed. I think we should probably make sure the error includes the startTime in millis and endTime in millis to tell the user about invalid times. Also, we should ideally include the valid start/end times in millis from TimeUtils.

Reply:
It will be printed in human readable format, which is better IMO because user can easily figure out their timestamp is invalid. I'll add start/end time in millis as well.
E.g. `Invalid segment start/end time: 2071-12-31T16:00:00.000/2071-12-31T16:00:00.000 (in millis: 3218832000000/3218832000000) for time column: date`

Reply:
Sounds good.